### ISSUE TITLE
be able to fetch remote changes of any branch

### ISSUE THREAD

**Issue by demarey** [pharo-vcs/iceberg#920]
Currently, it is possible to fetch changes from the upstream of the current branch but there is no way to fetch changes from a remote to your local repository of any branch.

**Comment by guillep** [comment]
Let me rephrase what I understand from your issue.

1. you have local branches A and B
2. your working copy is on A
3. remotely, there are new changes on B


- Do you want to "pull" changes from "remote/B" to "local/B" without checking out "local/B"?

This is not possible right now and we think this can cause more problems than solutions. 
Because pulling changes provokes always merge. In the happy cases, the merge will be a fast forward without conflicts, but in the case it is not you will need to solve conflicts, and that requires to modify code. Even, you will want to be able to test it. So it is much better if you checkout "local/B" and then pull, solve your conflicts, and finally go back to "local/A"

- or from "remote/B" to "local/A"?

If this is the case, this can be done with the merge dialog, I don't see the issue.

Even worst, using git from the command line you need to do the same to update a branch. You first checkout the branch to update and then pull. So in both cases, I don't think this is a valid issue.

**Comment by demarey** [comment]
I meant **fetch** changes from "remote/B" to "local/B" without checking out "local/B".
I would like to have this option to get commits that I do not have for an already defined remote. This way, I will then be able to choose the commit I want to merge for example.
